pkg/ingester/ingester.go

```diff
@@ -2214,7 +2212,9 @@ func (i *Ingester) createTSDB(_ context.Context, userID string, walReplayConcurr
 	// this will actually create the blocks. If there is no data (empty TSDB), this is a no-op, although
 	// local blocks compaction may still take place if configured.
 	level.Info(userLogger).Log("msg", "Running compaction after WAL replay")
-	err = db.Compact()
+	// Note that we want to let TSDB creation finish without being interrupted by eventual context cancellation,
```

Comment:
The code prior to the PR calls `db.Compact` without a context though, so passing `context.Background()` here only retains the current (pre-PR) behaviour.

If we want to make compaction cancellable, we can do it in a follow-up PR (since otherwise the PR does yet another thing)?

Reply:
> The code prior to the PR calls `db.Compact` without a context though, so passing `context.Background()` here only retains the current (pre-PR) behaviour.

Oh, I see. I would remove the comment then (I don't think it's true) and perhaps use `context.TODO()` as before.

> If we want to make compaction cancellable, we can do it in a follow-up PR (since otherwise the PR does yet another thing)?

Yes, let's do that in separate PR. Thanks.